### PR TITLE
[T4PUB-666] wait reboot sub to support multiple ecu

### PR DESCRIPTION
**Pull request by obi-t4**
# What is this PR?
This PR supports multiple ECU update/status requests.

# issue
~~https://tier4.atlassian.net/browse/T4PB-13539~~
https://tier4.atlassian.net/browse/T4PUB-666

# test
I confirm that update and status request for the two VirtualBox machines works properly.

### CODE REVIEW

File: app/ota_client_stub.py [hunk 1]
```diff
@@ -29,21 +31,50 @@ def __init__(self):
...
                         self._ota_client_call.update(request, secondary["ip_addr"]),
-                        name=secondary,  # register the task name with sub_ecu id
+                        # register the task name with sub_ecu id
+                        name=secondary["ecu_id"],
                     )
                 )
+            else:
+                logger.warning(f"{secondary['ecu_id']} was not found in the {request=}")
+
+        def can_reboot():
```
Comment (obi-t4) [inline app/ota_client_stub.py:58]: If all secondary ecus are SUCCESS, this function returns true.

**Comment by Bodong-Yang** [comment]
Please let me confirm the scenario:
1. mainECU dispatches ota-update request to all subECU, and subECU dispatches ota-update request to subsub...ECU, recursively, in a top-down fashion;
2. Only when all the child ECUs are updated, the parent ECU can update itself, recursively in bottom-up fashion.

Is that right?

**Comment by Bodong-Yang** [comment]
Actually I think the implementation of the logic should be restricted in the grpc-server, not stepped into the ota-client(I can see that a closure function **can_reboot** is passed deep into the ota-client).

More specifically speaking, I think the status pulling should be implemented in the grpc-server side(in ota-client-service), the scenarios may as follow:
(in ota_client_stub)
1. mainECU dispatches the ota-update requests to all subECU,
2. wait for the immediately responses from all subECUs non-blocking, confirm that all subECUs receive the request,
3. mainECU responses to the IoT agent non-blocking,
4. mainECU lauches a background tasks to pulling all subECUs and wait for updating itself.

**Comment by obi-t4** [comment]
> mainECU dispatches ota-update request to all subECU, and subECU dispatches ota-update request to subsub...ECU, recursively, in a top-down fashion;
Only when all the child ECUs are updated, the parent ECU can update itself, recursively in bottom-up fashion.

Right.
the parent ECU start updating itself, and wait reboot until all the child has been finished.

**Comment by obi-t4** [comment]
It is better to start updating for main ecu if update is requested, but the reboot should be waited until all children are finished.

**Comment by obi-t4** [comment]
I'd like to close this PR since #97 was merged.

**Pull request closed without merging**